src/backend/expungeservice/models/helpers/record_merger.py

```diff
@@ -16,14 +26,108 @@ def merge(
                 charge_id_to_time_eligibilities[k].add(v)
         record = ambiguous_record[0]
-        for charge in record.charges:
-            charge.expungement_result.time_eligibility = RecordMerger.merge_time_eligibilities(
-                charge_id_to_time_eligibilities[charge.id]
-            )
-        return record
+        new_case_list = []
+        for case in record.cases:
+            new_charges = []
+            for charge in case.charges:
+                time_eligibilities = charge_id_to_time_eligibilities.get(charge.id)
+                merged_type_eligibility = RecordMerger.merge_type_eligibilities(charge.id, case.charges)
+                merged_time_eligibility = RecordMerger.merge_time_eligibilities(time_eligibilities)
+                charge_eligibility = RecordMerger.compute_charge_eligibility(time_eligibilities)
+                expungement_result = ExpungementResult(
+                    type_eligibility=merged_type_eligibility,
+                    time_eligibility=merged_time_eligibility,
+                    charge_eligibility=charge_eligibility,
+                )
+                new_charge: Charge = replace(charge, expungement_result=expungement_result)
+                new_charges.append(new_charge)
+            new_case = replace(case, charges=new_charges)
+            new_case_list.append(new_case)
+        return replace(record, cases=new_case_list)
 
     @staticmethod
-    def merge_time_eligibilities(time_eligibilities: Set[TimeEligibility]):
-        return (
-            time_eligibilities.pop()
-        )  # TODO: Fix stub which assumes set is always of size 1 and actually do a meaningful merge.
+    def merge_type_eligibilities(charge_id: str, charges: List[Charge]) -> TypeEligibility:
+        same_charges = list(filter(lambda charge: charge.id == charge_id, charges))
+        status = RecordMerger.compute_type_eligibility_status(same_charges)
+        reason = ", ".join([charge.type_eligibility.reason for charge in same_charges])
+        return TypeEligibility(status=status, reason=reason)
+
+    @staticmethod
+    def compute_type_eligibility_status(charges: List[Charge]):
+        if all([charge.type_eligibility.status == EligibilityStatus.ELIGIBLE for charge in charges]):
+            return EligibilityStatus.ELIGIBLE
+        elif all([charge.type_eligibility.status == EligibilityStatus.INELIGIBLE for charge in charges]):
+            return EligibilityStatus.INELIGIBLE
+        else:
+            return EligibilityStatus.NEEDS_MORE_ANALYSIS
+
+    @staticmethod
+    def merge_time_eligibilities(time_eligibilities: Optional[Set[TimeEligibility]]) -> Optional[TimeEligibility]:
+        if time_eligibilities:
+            status = RecordMerger.compute_time_eligibility_status(time_eligibilities)
+            reason = ", ".join([time_eligibility.reason for time_eligibility in time_eligibilities])
+            date_will_be_eligible = date.max  # TODO: Fix
```

Comment:
@wittejm I believe the frontend does not use this value.

Reply:
Yep, I recall putting it in there thinking "hmm it seems like the frontend should probably be using this at some point" but that doesn't seem like good reasoning in retrospect.